src/Confluent.Kafka/ConsumerBuilder.cs

```diff
@@ -275,27 +290,41 @@ public ConsumerBuilder(IEnumerable<KeyValuePair<string, string>> config)
         }
 
         /// <summary>
-        ///     This handler is called when a new consumer group partition assignment has been received
-        ///     by this consumer.
-        ///     
-        ///     Note: corresponding to every call to this handler there will be a corresponding call to
-        ///     the partitions revoked handler (if one has been set using SetPartitionsRevokedHandler).
+        ///     Specify a handler that will be called when a new consumer group partition assignment has
+        ///     been received by this consumer.
         ///
         ///     The actual partitions to consume from and start offsets are specified by the return value
-        ///     of the handler. This set of partitions is not required to match the assignment provided
-        ///     by the consumer group, but typically will. Partition offsets may be a specific offset, or
-        ///     special value (Beginning, End or Unset). If Unset, consumption will resume from the
-        ///     last committed offset for each partition, or if there is no committed offset, in accordance
-        ///     with the `auto.offset.reset` configuration property.
+        ///     of the handler. Partition offsets may be a specific offset, or special value (Beginning, End
+        ///     or Unset). If Unset, consumption will resume from the last committed offset for each
+        ///     partition, or if there is no committed offset, in accordance with the `auto.offset.reset`
+        ///     configuration property.
+        ///
+        ///     Kafka supports two rebalance protocols: EAGER (range and roundrobin assignors) and
+        ///     COOPERATIVE ("incremental") (sticky-cooperative assignor).
+        ///
+        ///     ## EAGER Rebalancing
+        ///
+        ///     The set of partitions returned from your handler may differ from that provided by the
+        ///     group (though they should typically be the same). These partitions are the
+        ///     entire set of partitions to consume from. There will be exactly one call to the
+        ///     partitions revoked or partitions lost handler (if they have been set using
+        ///     SetPartitionsRevokedHandler / SetPartitionsLostHandler) corresponding to every call to
+        ///     this handler.
+        ///
+        ///     ## COOPERATIVE (Incremental) Rebalancing
+        ///
+        ///     The set of partitions returned from your handler must match that provided by the
```

Comment:
Don't like this constraint, is there an actual reason for it?

Reply:
the reason is we don't want to encourage people to do it - the logic to track it isn't super hard, but it's getting a little involved, and it's just a bad API IMO.

Reply:
because to realize a ctrl topic, they don't need to adjust the group assignment in the rebalance cb (it will happen outside rebalance_cb), and I hate that this is possible.